collectors/scm/gitlab/src/main/java/com/capitalone/dashboard/gitlab/GitlabRequestsResponseMapper.java

```diff
@@ -0,0 +1,218 @@
+package com.capitalone.dashboard.gitlab;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.joda.time.DateTime;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestOperations;
+
+import com.capitalone.dashboard.gitlab.model.GitlabCommitStatus;
+import com.capitalone.dashboard.gitlab.model.GitlabNote;
+import com.capitalone.dashboard.gitlab.model.GitlabRequest;
+import com.capitalone.dashboard.misc.HygieiaException;
+import com.capitalone.dashboard.model.Comment;
+import com.capitalone.dashboard.model.CommitStatus;
+import com.capitalone.dashboard.model.GitRequest;
+import com.capitalone.dashboard.model.Review;
+
+@Component
+public class GitlabRequestsResponseMapper {
+	private static final Log LOG = LogFactory.getLog(GitlabRequestsResponseMapper.class);
+
+	private RestOperations restOperations;
+	private GitlabUrlUtility gitlabUrlUtility;
+	private String apiToken;
+
+	public void init(GitlabUrlUtility gitlabUrlUtility, RestOperations restOperations) {
+		this.gitlabUrlUtility = gitlabUrlUtility;
+		this.restOperations = restOperations;
+	}
+
+	public List<GitRequest> map(GitlabRequest[] gitlabRequests, String repoUrl, String branch, String apiToken,
+			Map<Long, String> mrCloseMap) throws HygieiaException {
+		List<GitRequest> requests = new ArrayList<GitRequest>();
+
+		String targetBranch = branch;
+		if (StringUtils.isBlank(targetBranch)) {
+			targetBranch = "master";
+		}
+
+		this.apiToken = apiToken;
+
+		String[] orgAndRepoName = null;
+
+		for (GitlabRequest gitlabRequest : gitlabRequests) {
+			LOG.debug("targetBranch: " + targetBranch);
+			LOG.debug("gitlabRequest.getTargetBranch(): " + gitlabRequest.getTargetBranch());
+			if (StringUtils.equals(targetBranch, gitlabRequest.getTargetBranch())) {
+				GitRequest request = map(repoUrl, targetBranch, gitlabRequest);
+				if (null == orgAndRepoName) {
+					orgAndRepoName = this.gitlabUrlUtility.getOrgAndRepoName(gitlabRequest.getWebUrl());
+				}
+				request.setOrgName(orgAndRepoName[0]);
+				request.setRepoName(orgAndRepoName[1]);
+				requests.add(request);
+
+				boolean stop = (!MapUtils.isEmpty(mrCloseMap) && mrCloseMap.get(request.getUpdatedAt()) != null)
```

Comment:
Can you update this to make it more meaningful? For better readability?

Reply:
https://github.com/Hygieia/Hygieia/pull/2755/commits/3cbb00eff8b1daf70388ea69d7c1d03eb0682045#diff-570da31f834a8e819550dc23f5059bc9R70